core/objects.go

```diff
@@ -152,6 +154,74 @@ type Challenge struct {
 	Nonce string `json:"nonce,omitempty"`
 }
 
+// Check the sanity of a challenge object before issued to the client (completed = false)
+// and before validation (completed = true).
+func (ch Challenge) IsSane(completed bool) bool {
+	switch ch.Type {
+	case ChallengeTypeSimpleHTTPS:
+		// check extra fields aren't used
+		if ch.R != "" || ch.S != "" || ch.Nonce != "" {
+			return false
+		}
+
+		if completed {
+			// see if ch.Path starts with /.well-known/acme-challenge/
+			if ch.Path == "" || !strings.HasPrefix(ch.Path, "/.well-known/acme-challenge/") {
```

Comment:
It's my understanding that `Path` is a field that only the client should populate, i.e. a SimpleHTTPS challenge shouldn't come from the `PA` with anything in `Path`, is this not right?